src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,728 @@
+
+                var endpointUuid = Uuid.Generate().Value;
+                var endpointId = endpointUuid.Split(":").Last();
+                var repImagingStudyText = new Text("generated",
+                    "<div xmlns=\"http://www.w3.org/1999/xhtml\">Unstructured data can be sent here</div>");
+                var diagRepImagingStudyRep =
+                    new DiagnosticReportImagingStudyRepresentation(imagingStudyUuid, GetImagingStudyResource(endpointId,
+                        reportAsImage,
+                        repImagingStudyText, imagingStudyId));
+                representations.Add(diagRepImagingStudyRep);
+                
+                var diagRepEndpointRep = new DiagnosticReportEndpointRepresentation(endpointUuid,
+                    GetEndpointResource(reportAsImage, endpointId, repImagingStudyText));
+                representations.Add(diagRepEndpointRep);
+            }
+
+            if (!representations.Any())
+            {
+                return Option.None<DiagnosticReportResponse>();
+            }
+
+            var diagnosticReportResponse = new DiagnosticReportResponse
+            {
+                Entry = representations,
+                Id = Uuid.Generate().Value.Split(":").Last(),
+                ResourceType = "Bundle",
+                Type = "collection"
+            };
+            return Option.Some<DiagnosticReportResponse>(diagnosticReportResponse);
+        }
+
+        private static EndpointResource GetEndpointResource(DiagnosticReportAsImage reportAsImage,
+            string endpointId, Text repImagingStudyText)
+        {
+            var connectionType =
+                new ConnectionType("http://terminology.hl7.org/CodeSystem/endpoint-connection-type",
+                    "dicom-wado-rs");
+            return new EndpointResource(HiType.Endpoint, endpointId, repImagingStudyText, "active",
+                connectionType, new List<PayloadType> {new PayloadType(reportAsImage.PayloadType)},
+                new List<string> {reportAsImage.PayloadMimeType},
+                reportAsImage.ReportUrl);
+        }
+
+        private static ImagingStudyResource GetImagingStudyResource(string endpointId,
+            DiagnosticReportAsImage reportAsImage, Text repImagingStudyText, string imagingStudyId)
+        {
+            var subjectImagingStudy = new SubjectDiagReport(reportAsImage.SubjectReference);
+            return new ImagingStudyResource(HiType.ImagingStudy, imagingStudyId,
+                repImagingStudyText, subjectImagingStudy, "available", reportAsImage.StudyStartDate,
+                reportAsImage.NumberOfSeries,
+                reportAsImage.NumberOfInstances,
+                new List<Endpoint> {new Endpoint("Endpoint" + "/" + endpointId)});
+        }
+
+        private static DiagnosticReportImage GetDiagRepResource(DiagnosticReportAsImage reportAsImage,
+            string imagingStudyUuid, string imagingStudyId, string id, string patientName)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'imagingStudyUuid'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)